### DISCUSSION TITLE
Welcome to Spingular7 Discussions! Present yourself to the Team.

### DISCUSSION THREAD

**Discussion by Tonterias** [Tonterias/SpingularJhipsterV7#1] [General]
## 👋 Welcome to [Spingular V7 Basic](https://www.spingular.com/#/)

First, I would like to summarize Spingulars' objectives and let me know if my idea makes sense to you: 

1.  If we can make an open source CMS made with the latest technology of Jhipster that others can use & 
2. If we can document the process while we do it (on GitHub &  Spingulars' Posts) so it is a guide for others who are learning Jhipster (or want to join later on) & 
3. If we create a community of people who want to learn/share with others in the process... I will be very satisfied!

If you are asking yourself: how much time this is going to take from me? As much or as little as you want. All I ask is to deliver what each one agrees to do.

[Video instructions on how to use GitHub Discussions in case it is new for you](https://githubuniverse.com/Building-your-community-with-GitHub-Discussions/)

  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and be open-minded. Remember that this is a community we build together 💪.

### To get started, comment below with an introduction of yourself and tell us about what you do with this community. 
Please, copy and answer these questions as well: (Y/N) or (Beginner/Intermediate/Advanced). I will do it first:

- Jhipster:  (None/Beginner/Intermediate/Advanced)
- Java/ Spring:  (None/Beginner/Intermediate/Advanced)
- Angular: (None/Beginner/Intermediate/Advanced)
- Database: Postgres - (None/Beginner/Intermediate/Advanced)
- MapStruct: (None/Beginner/Intermediate/Advanced)
- GitHub:  (Beginner/Intermediate/Advanced)
- Testing :  (Beginner/Intermediate/Advanced)
- Deployment: Docker  (Beginner/Intermediate/Advanced) & CD/CI: Beginner & Heroku:  (Beginner/Intermediate/Advanced)
- Add others you consider important

**If you're new and want to join Spingular, please let me know (@spingularpress or spingular at gmail) I will send you an invitation.** 

We will start using discussions at Github to agree any general questions about the project, such as, review the JDL file, setup the project, discuss a new Discussion category to be added and see which functionalities we approve. Then,

- Please, use [Issues](https://github.com/Tonterias/SpingularJhipsterV7/issues) to alert about problems in the app
- Work with [Pull Requests](https://github.com/Tonterias/SpingularJhipsterV7/pulls) to update code.
- For documentation: visit the [Wiki](https://github.com/Tonterias/SpingularJhipsterV7/wiki)
- We will try to use [Projects](https://github.com/Tonterias/SpingularJhipsterV7/projects) to see the tasks and timetable

If you give a Star to the project, we will know that you read everything :-). Great!

**Comment by Tonterias** [comment]
I will go first. My name is Miguel, but you can call me Mike (since everybody pronounces that perfect, not like Miguel ;-). I live in Spain (Madrid). I'm not a programmer, but I have been working in programming projects for a long time and I started programming so I could talk with the programmers in the project and understand them.

- Jhipster: Intermediate
- Java/ Spring: Intermediate
- Angular: Intermediate
- Database: Postgres - Intermediate
- MapStruct: Intermediate
- GitHub: Intermediate
- Testing : Beginner
- Deployment: Docker Beginner & CD/CI: Beginner & Heroku: Beginner
- Add others you consider important

**Reply by silverwind85** [reply]
HI, my name is Slawek I come form in Poland . I'm fullstack developer. 

- Jhipster: Beginner
- Java/ Spring: Advanced
- Angular: Beginner
- Database: Postgres - Intermediate
- MapStruct: Beginner
- GitHub: Intermediate
- Testing : Beginner
- Deployment: Docker (Beginner/Intermediate/Advanced) & CD/CI: Beginner & Heroku: Beginner

**Comment by sandeshdahake** [comment]
My name is Sandesh. I am an Indian who lives in Paris. I am an electronics engineer by education but my IKIGAI directed me towards programming. 

- Jhipster: Beginner
- Java/ Spring: Advanced
- Angular: None
- Database: Postgres - Intermediate
- MapStruct: Advanced
- GitHub: Intermediate 
- Testing: Intermediate
- Deployment: Docker Beginner & CD/CI: Beginner & AWS: Beginner
- REST - Advanced 
- GraphQL - Beginner

**Comment by cchunduri** [comment]
My Name is Chaitanya. I live in Hyderabad, India. I am having six-plus years of experience in the programming area. I am a full-stack engineer can work on the backend, frontend and Android applications too. At present, my focus is to write clean code, by following solid principles, design patterns and Test-Driven Development.

- Jhipster: Intermediate
- Java/ Spring: Advanced
- Vanilla JS: Advanced
- Angular: Intermediate
- ReactJS: Advanced
- Database: Postgres - Intermediate
- MapStruct: Intermediate
- GitHub: Advanced
- Testing: Advanced (Considering writing Selenium Scripts for automation)
- CD/CI: Jenkins - Intermediate,  Docker - Intermediate,  Kubernetes - Beginner 
- Heroku: Intermediate

**Comment by arnaugarcia** [comment]
Hello everyone, my name is Arnau (Arnold in english) and I live in Spain (Barcelona). I've at least 3 years of experience with JHipster and 6+ of full-stack programming... mainly in Java and Angular.

Jhipster: Advanced
Java/ Spring: Advanced
Angular: Advanced

Databases: 
- MySQL - Intermediate 
- MongoDB - Intermediate

MapStruct: Intermediate
GitHub: Advanced
Testing : Intermediate

Deployment: 
- Docker Intermediate 
- CD/CI: Beginner 
- AWS: Intermediate

**Comment by Tonterias** [comment]
**Thanks for your patience!**

Let’s start and the rest can join later and fill the presentation.

We will communicate around here, but I want to open a chat so we can have instant communication, if needed. Any preference? I will create it in Hang-outs and include you, if no one has a problem. We should seldom use it!

Looking at the knowledge of the group, I think that the best approach is to load the time-consuming tasks on the ones who know less (probably me) and keep the ones who know more dealing with problems and difficult tasks. Nonetheless, if you have time and want to contribute more, raise your hand please!

- Release 7.0.0-beta.1 is out so the final v7 must be around. In the meantime, you might take a look at the [JDL file that I pasted on the next discussion](https://github.com/Tonterias/SpingularJhipsterV7/discussions/3) and make it shine with your comments and ideas. I will try to create one discussion for each tasks, so we have documentation of the process.

- I have created a [warmup git branch](https://github.com/Tonterias/SpingularJhipsterV7/blob/warmup/jdl/SpingularV7.jh) to play around with the JDL file and test data for H2 local DB. I made it with JHv6.10.1 so you can clone it and make it work locally. 

**#TODO**: We will have to decide on the [yo-rc.json](https://github.com/Tonterias/SpingularJhipsterV7/blob/warmup/.yo-rc.json) decisions.

![Spingular1](https://user-images.githubusercontent.com/11610877/105847893-8a252280-5fde-11eb-8529-a3293c421147.png)

•	@sandeshdahake: If you are new to Jhipster: [JDL is explained here](https://www.jhipster.tech/jdl/entities-fields#-jhipster-domain-language-jdl---entities) and [here](https://www.jhipster.tech/jdl/getting-started#-jhipster-domain-language-jdl---getting-started). Please,  if you have any questions, let me know privately in Hangout. [I recommend Full Stack Development with JHipster - Second Edition](https://www.packtpub.com/product/full-stack-development-with-jhipster-second-edition/9781838824983). It is really good to start and save time, thought you can do it with Jhipster site and take a look at the old [Spingular version](https://www.spingular.com/#/post/10/view), too.

•	Here is the [data file to fill the H2 database in development](https://github.com/Tonterias/SpingularJhipsterV7/blob/warmup/db_testData/InsertsInPOSTGRESQLDatabaseSpingularV7.txt). All you need is paste and run.

![Spingular2](https://user-images.githubusercontent.com/11610877/105848391-2fd89180-5fdf-11eb-9c22-8fb394d1b24f.png)


**#HELP: Please @cchunduri, let me know how you would organize the git structure (or where to find best practices to follow). 
@arnaugarcia, let me know how you would make a CD/CI pipeline for this or if I'm missing something important from the start.**

NEXT STEP: We can fix the JDL until Jhipster v7 is realeased and we can start moving everything into the new version.

**Reply by cchunduri** [reply]
@Tonterias sure I can help with that?
My comments about the Git structure are [here](https://github.com/Tonterias/SpingularJhipsterV7/discussions/6)
Check them out!

**Reply by Tonterias** [reply]
Thanks Chaitanya. I will add my comments and the others can add theirs (if they have knowledge or learn ;-). 

Please, when you answer try to be gentle with everybody and add your thoughts if you really have a better (best practice) alternative. 

Let's keep it simple and move forward! 

This is a lot about getting to know each other (and make friends too ;-).

**Comment by burkinabe** [comment]
Hi, my name is Abdoul Karim SAVADOGO from Burkina Faso, West Africa. I'm a Full Stack Developer with 3 years of experience. I have been working with Jhipster since 2018. 

Jhipster: Advanced
Java/ Spring: Intermediate
Angular: Intermediate
Database: Postgres - Intermediate
MapStruct: Intermediate
GitHub: Intermediate
Testing : Beginner
Deployment: Docker Beginner & CD/CI: Beginner & Heroku: Beginner

**Reply by AdalWay** [reply]
Hi everyone, my name is Adalberto Cuevas, currently living in Canada but originally from the Dominican Republic. I have like 2 years of experience working with Jhipster.

Jhipster: Intermediate
Java/ Spring: Intermediate
Database: Postgres - Intermediate
MapStruct: Intermediate
GitHub: Intermediate

Deployment:

Docker Intermediate
CD/CI: Beginner
AWS: Intermediate

**Comment by mgarayarc** [comment]
Hi everybody, I'm another Mike (or Miguel Garayar as you wish ^_^) and I'm excited to be part of this project. I'm from Peru and I've being a programmer for around 9 years but then I had an opportunity to see the architecture and business side of the company I work for, and when I realized I wanted to come back to programming there have being passed 5 more years. I hope I can help you guys with my modest skills:

Jhipster: Beginner
Java/ Spring: Advanced / Intermediate
Angular: Beginner
Typescript: Beginner
Database: Postgres - Beginner
MapStruct: Beginner
GitHub: Beginner
Testing : Beginner
Azure Solution Architecture: Intermediate

**Reply by Tonterias** [reply]
Hola Miguel,

You can be Miguel and we wont get confused ;-)

I have been waiting a little for version 7 of Jhipster to settle down, because I had a few problems. I have [test it](https://github.com/Tonterias/spingularTestV7/commits/main) and now it looks good to me. The test data is updated.

So let's move forward! I'm going to contact to the rest and wake the project up.

In the meantime, please take a look at the [functionalities](https://github.com/Tonterias/SpingularJhipsterV7/discussions/3) in the .JH file and take a look at the [GIT strategy](https://github.com/Tonterias/SpingularJhipsterV7/discussions/6) to be on the same page as the rest and we can start working on [the development](https://github.com/Tonterias/SpingularJhipsterV7/discussions/4)

Happy that you join us.

**Comment by seyeadamaUASZ** [comment]
Hi,my name is Adama Seye and i'm a fullstack developer. So my specialities are:
Spring boot:advanced
Angular:advanced
JHipster:Beginner

**Reply by cchunduri** [reply]
Hi Adama Seye,

Welcome to team.

**Comment by seyeadamaUASZ** [comment]
when the next sessions?

**Reply by Tonterias** [reply]
Hi Adama, 

Please send me an email to spingular @ gmail or to my email and we I will explain the status of the project now. 

Welcome to Spingular
